cmd/zdb/zdb.c

```diff
@@ -1967,17 +1967,49 @@ dump_dedup_ratio(const ddt_stat_t *dds)
 static void
 dump_ddt_log(ddt_t *ddt)
 {
+	if (ddt->ddt_version != DDT_VERSION_FDT ||
+	    !(ddt->ddt_flags & DDT_FLAG_LOG))
+		return;
+
 	for (int n = 0; n < 2; n++) {
 		ddt_log_t *ddl = &ddt->ddt_log[n];
 
-		uint64_t count = avl_numnodes(&ddl->ddl_tree);
-		if (count == 0)
-			continue;
+		char flagstr[64] = {0};
+		if (ddl->ddl_flags > 0) {
+			flagstr[0] = ' ';
+			int c = 1;
+			if (ddl->ddl_flags & DDL_FLAG_FLUSHING)
+				c += strlcpy(&flagstr[c], " FLUSHING",
+				    sizeof (flagstr) - c);
+			if (ddl->ddl_flags & DDL_FLAG_CHECKPOINT)
+				c += strlcpy(&flagstr[c], " CHECKPOINT",
+				    sizeof (flagstr) - c);
+			if (ddl->ddl_flags &
+			    ~(DDL_FLAG_FLUSHING|DDL_FLAG_CHECKPOINT))
+				c += strlcpy(&flagstr[c], " UNKNOWN",
+				    sizeof (flagstr) - c);
+			flagstr[1] = '[';
+			flagstr[c++] = ']';
+		}
 
-		printf(DMU_POOL_DDT_LOG ": %lu log entries\n",
-		    zio_checksum_table[ddt->ddt_checksum].ci_name, n, count);
+		uint64_t count = avl_numnodes(&ddl->ddl_tree);
 
-		if (dump_opt['D'] < 4)
+		printf(DMU_POOL_DDT_LOG ": flags=0x%02x%s; obj=%lu; "
+		    "len=0x%lx; txg=%lu; entries=%lu\n",
+		    zio_checksum_table[ddt->ddt_checksum].ci_name, n,
+		    ddl->ddl_flags, flagstr, ddl->ddl_object,
+		    ddl->ddl_length, ddl->ddl_first_txg, count);
+
+		if (ddl->ddl_flags & DDL_FLAG_CHECKPOINT)
+			printf("    checkpoint: "
+			    "%016lx:%016lx:%016lx:%016lx:%016lx\n",
+			    ddl->ddl_checkpoint.ddk_cksum.zc_word[0],
```

Comment:
I think `%016lx` won't do the trick on 32bit platforms.

Reply:
Sigh. There are too many things to remember heh.

Redone following the pattern elsewhere ie `%016llx` and cast to `(u_longlong_t)`.